apps/xmtp.chat/src/components/Messages/Message.tsx

```diff
@@ -1,44 +1,35 @@
 import { Box, Flex, Paper, Stack, Text } from "@mantine/core";
-import type { Client, DecodedMessage } from "@xmtp/browser-sdk";
+import type { Client, Conversation, DecodedMessage } from "@xmtp/browser-sdk";
+import {
+  ContentTypeTransactionReference,
+  type TransactionReference,
+} from "@xmtp/content-type-transaction-reference";
+import {
+  ContentTypeWalletSendCalls,
+  type WalletSendCallsParams,
+} from "@xmtp/content-type-wallet-send-calls";
 import { intlFormat } from "date-fns";
-import { useNavigate, useOutletContext } from "react-router";
+import { useOutletContext } from "react-router";
 import { shortAddress } from "@/helpers/address";
 import { nsToDate } from "@/helpers/date";
-import classes from "./Message.module.css";
 import { MessageContent } from "./MessageContent";
+import { TransactionReferenceUI } from "./TransactionReference";
+import { WalletSendCallsUI } from "./WalletSendCalls";
 
 export type MessageProps = {
   message: DecodedMessage;
+  sendMessage: Conversation["send"];
 };
 
-export const Message: React.FC<MessageProps> = ({ message }) => {
+export const Message: React.FC<MessageProps> = ({ message, sendMessage }) => {
   const { client } = useOutletContext<{ client: Client }>();
   const isSender = client.inboxId === message.senderInboxId;
   const align = isSender ? "right" : "left";
-  const navigate = useNavigate();
 
   return (
     <Box px="md">
       <Flex justify={align === "left" ? "flex-start" : "flex-end"}>
-        <Paper
-          p="md"
-          withBorder
-          shadow="md"
-          maw="80%"
-          tabIndex={0}
-          onKeyDown={(e) => {
-            if (e.key === "Enter") {
-              void navigate(
-                `/conversations/${message.conversationId}/message/${message.id}`,
-              );
-            }
-          }}
-          className={classes.root}
-          onClick={() =>
-            void navigate(
-              `/conversations/${message.conversationId}/message/${message.id}`,
-            )
-          }>
```

Comment:
this removes a key feature of the dev tool and needs to be preserved. i believe you should be able to prevent the click event propagation of the links in the content to not trigger the modal.

Reply:
ok, will try it out.